### PR TITLE
Copilot Chat: Fix redirect URI update in deploy scripts

### LINKED ISSUE #1788: copilot-chat-app azure deploy. Users who are not from the current tenant cannot log in properly.
**Describe the bug**
copilot-chat-app azure deploy. Users who are not from the current tenant cannot log in properly.
After logging in, it will return to the page before the login, and report the following error.
`AADSTS9002326: Cross-origin token redemption is permitted only for the 'Single-Page Application' client-type`
The Azure deployment was done using the script below.
[semantic-kernel](https://github.com/microsoft/semantic-kernel/tree/main)/[samples](https://github.com/microsoft/semantic-kernel/tree/main/samples)/[apps](https://github.com/microsoft/semantic-kernel/tree/main/samples/apps)/[copilot-chat-app](https://github.com/microsoft/semantic-kernel/tree/main/samples/apps/copilot-chat-app)/[deploy](https://github.com/microsoft/semantic-kernel/tree/main/samples/apps/copilot-chat-app/deploy)
/deploy-webapp.ps1

The application was registered as a Single Page Application (SPA) type
I suspect that the problem was caused by the command below. #1711 
```
Write-Host "Updating AAD App registration..."
az ad app update --id $ApplicationClientId --web-redirect-uris "https://$webappUrl"
```

**Expected behavior**
Users from different tenants can also log in normally.

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Windows 11
 -Browser:  Edge

**Additional context**

### PR DESCRIPTION
**Pull request by gitri-ms**
### Motivation and Context

Fixes #1788

### Description

Correct the part of the deploy scripts that updates the redirect URI. The previously used command only works for the "web" platform type, not "SPA" which Copilot Chat uses. Azure CLI does not support the latter, so using `az rest` instead.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/deploy/deploy-webapp.ps1 [hunk 1]
```diff
@@ -89,13 +89,6 @@ if ($LASTEXITCODE -ne 0) {
     exit $LASTEXITCODE
 }
 
-Write-Host "Updating AAD App registration..."
```
Comment (glahaye) [inline samples/apps/copilot-chat-app/deploy/deploy-webapp.ps1:92]: There might be a way to do this despite the limitations to "az ad app update":
https://stackoverflow.com/questions/68744719/azure-create-app-registration-with-replyurlswithtype

Comment (gitri-ms) [reply samples/apps/copilot-chat-app/deploy/deploy-webapp.ps1:92]: Yes, this does seem to work. It overwrites whatever redirectUri is already there, but I think that's ok. I've got it working for PowerShell, doing bash now..

**Review by amsacha — DISMISSED** [review]

File: samples/apps/copilot-chat-app/deploy/deploy-webapp.sh [hunk 2]
```diff
@@ -131,6 +122,14 @@ if [[ "$CORS_RESULT" == "null" ]]; then
     az webapp cors add --name $WEB_API_NAME --resource-group $RESOURCE_GROUP --subscription $SUBSCRIPTION --allowed-origins $origin
 fi
 
+echo "Updating redirect URI in AAD app registration to '$ORIGIN'..."
+eval OBJECT_ID=$(az ad app show --id $APPLICATION_ID | jq '.id')
+az rest \
+    --method PATCH \
+    --uri "https://graph.microsoft.com/v1.0/applications/$OBJECT_ID" \
+    --headers 'Content-Type=application/json' \
+    --body "{spa:{redirectUris:['$ORIGIN']}}"
```
Comment (gitri-ms) [inline samples/apps/copilot-chat-app/deploy/deploy-webapp.sh:131]: This command doesn't seem to be working, trying to figure out what's wrong... no errors, but i'm not seeing the app registration updating. The object id is parsed correctly.

File: samples/apps/copilot-chat-app/deploy/README.md [hunk 3]
```diff
@@ -98,7 +98,11 @@ npm install -g @azure/static-web-apps-cli
 ./deploy-webapp.sh --subscription {YOUR_SUBSCRIPTION_ID} --resource-group rg-{YOUR_DEPLOYMENT_NAME} --deployment-name {YOUR_DEPLOYMENT_NAME} --application-id {YOUR_APPLICATION_ID}
 ```
 
-Your CoplitChat application is now deployed!
+## Update your redirect URI
+Go back to your AAD application registration and add your frontend's URL as an additional redirect URI.  
+_**Note:** This step must be done via the Azure portal, as the Azure CLI does not currently support it for the SPA platform type._
+
```
Comment (gitri-ms) [inline samples/apps/copilot-chat-app/deploy/README.md:104]: Remove

**Review by glahaye — APPROVED** [review]

**Review by teresaqhoang — DISMISSED** [review]

**Review by shawncal — DISMISSED** [review]

File: samples/apps/copilot-chat-app/deploy/deploy-webapp.ps1 [hunk 4]
```diff
@@ -89,19 +89,20 @@ if ($LASTEXITCODE -ne 0) {
...
 }
 
+Write-Host "Updating redirect URI in AAD app registration to '$webappUrl'..."
+$objectId = (az ad app show --id $ApplicationClientId | ConvertFrom-Json).id
+az rest `
+    --method PATCH `
+    --uri "https://graph.microsoft.com/v1.0/applications/$objectId" `
+    --headers 'Content-Type=application/json' `
+    --body "{spa:{redirectUris:['$origin']}}"
+
```
Comment (gitri-ms) [inline samples/apps/copilot-chat-app/deploy/deploy-webapp.ps1:105]: This currently overwrites any URIs. Need to append instead

**Review by glahaye — APPROVED** [review]

**Pull request merged**